functions-app/report-psychometrician-generate-staging-file/service/data.service.js

```diff
@@ -0,0 +1,243 @@
+    const baseFilename = 'ps-staging-1.csv'
+    const fileNameWithPath = `${directory}${path.sep}${baseFilename}`
+    const sql = `SELECT
+                     chk.id AS checkId,
+                     chk.checkCode,
+                     chk.checkForm_id as checkFormId,
+                     chk.createdAt AS checkCreatedAt,
+                     chk.isLiveCheck,
+                     chk.pupilLoginDate,
+                     cs.code AS checkStatus,
+                     cs.description AS checkStatusDescription,
+                     prr.code AS restartCode,
+                     (SELECT COUNT(id)
+                        FROM [mtc_admin].[pupilRestart] pr
+                       WHERE pr.pupil_id = chk.pupil_id
+                         AND pr.createdAt < chk.createdAt
+                         AND pr.isDeleted = 0) AS restartCount,
+                     ac.code AS attendanceCode,
+                     p.foreName,
+                     p.middleNames,
+                     p.lastName,
+                     p.dateOfBirth,
+                     p.upn,
+                     p.gender,
+                     cw.id AS checkWindowId,
+                     cw.name AS checkWindowName,
+                     s.name AS schoolName,
+                     s.urn AS schoolUrn,
+                     s.estabCode AS schoolEstabCode,
+                     s.leaCode AS schoolLeaCode,
+                     s.urlSlug as schoolGuid,
+                     cf.name AS checkFormName
+
+                   FROM [mtc_admin].[pupil] p
+                        JOIN      [mtc_admin].[school] s ON (p.school_id = s.id)
+                        LEFT JOIN [mtc_admin].[check] chk ON (p.currentCheckId = chk.id)
+                        LEFT JOIN [mtc_admin].[checkForm] cf ON (chk.checkForm_id = cf.id)
+                        LEFT JOIN [mtc_admin].[checkStatus] cs ON (chk.checkStatus_id = cs.id)
+                        LEFT JOIN [mtc_admin].[checkWindow] cw ON (chk.checkWindow_id = cw.id)
+                        LEFT JOIN [mtc_admin].[attendanceCode] ac ON (p.attendanceId = ac.id)
+                        LEFT JOIN [mtc_admin].[pupilRestart] pr ON (pr.check_id = chk.id AND pr.isDeleted = 0)
+                        LEFT JOIN [mtc_admin].[pupilRestartReason] prr ON (prr.id = pr.pupilRestartReason_id)
+                  WHERE p.attendanceId IS NOT NULL
+                     OR p.checkComplete = 1;`
+    psychometricianDataService.setLogger(this.logger)
+    await psychometricianDataService.streamReport(fileNameWithPath, sql)
+    return fileNameWithPath
+  },
+  /**
+   * Dump all data required to produce the entire psychometrician report
+   * This will include check, pupil, checkForm, answers, school
+   * @param {String} $tmpDirectory - the directory to store the staging files in
+   *  For Unix this can be $TMP
+   *  For Azure Functions running windows this should be the web server root
+   * @return {Promise<void>}
+   */
+  dumpStage1File: async function dumpStage1File () {
```

Comment:
@jon-shipley is there a description documented somewhere about the stages and their purpose?
Shall I assume stage-1 is data dumping and stage-2 is writing to file based on the methods in this file?

Reply:
@GeorgeChatzigiannis - the jsdoc explains it